### PR TITLE
Added support for internal insecure TLS routes.

### PR DESCRIPTION
**Pull request by Bplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: lib/discovery/updater.go [hunk 1]
```diff
@@ -156,35 +160,50 @@ func (u *updater) onModifiedOrAddedEvent(serviceObj service, service serviceKey,
...
+		case httpScheme:
+			fallthrough
+		case httptlsScheme:
 			if hostMatcherOverride != "" {
 				foundRoute.nameMatcher = hostMatcherOverride
 			}
 
 			// We need to avoid specific ports if possible, since Golang removes the port from request.URL.Port() for default ports,
-			// even when user specifies http://<host>:80 or https://<host>:443 explicitly. As a result we convert default
+			// even when user specifies httpScheme://<host>:80 or https://<host>:443 explicitly. As a result we convert default
```
Comment (improbable-vova) [inline lib/discovery/updater.go]: Probably https -> httptlsScheme as well

File: lib/discovery/updater.go [hunk 2]
```diff
@@ -156,35 +160,50 @@ func (u *updater) onModifiedOrAddedEvent(serviceObj service, service serviceKey,
 			portMatcher: port.Port,
 		}
 
-		// TODO(bplotka): Add support for TLS between (even insecure for now).
-		if port.Name == "http" || strings.HasPrefix(port.Name, "http-") {
+		scheme := getScheme(port.Name)
+		switch scheme {
+		case httpScheme:
```
Comment (improbable-vova) [inline lib/discovery/updater.go]: Use multicase like `case httpScheme, httpTlsScheme`

File: lib/discovery/updater.go (hunk 1 above)
Comment (Bplotka) [reply lib/discovery/updater.go]: ah, good point.. my find & replace failure

File: lib/discovery/updater.go [hunk 3]
```diff
@@ -156,35 +160,46 @@ func (u *updater) onModifiedOrAddedEvent(serviceObj service, service serviceKey,
...
 			}
 
 			// We need to avoid specific ports if possible, since Golang removes the port from request.URL.Port() for default ports,
 			// even when user specifies http://<host>:80 or https://<host>:443 explicitly. As a result we convert default
 			// ports to NO port, which mean host-wide routing.
-			if port.Port == 80 {
+			if scheme == httpScheme && port.Port == 80 {
+				foundRoute.portMatcher = 0
+			}
+			if scheme == httptlsScheme && port.Port == 443 {
```
Comment (improbable-vova) [inline lib/discovery/updater.go]: I would like to move these checks to some func e.g.
```
if isDefaultPort(scheme, port.Port) {
    foundRoute.portMatcher = 0
}
```
but it's up to you

Comment (Bplotka) [reply lib/discovery/updater.go]: that makes sense

**Pull request merged**